src/rabbit_auth_backend_oauth2.erl

```diff
@@ -282,3 +318,39 @@ matching_scopes_without_prefix(Scopes, PrefixPattern) ->
             end
         end,
         Scopes).
+
+check_picklock(Scopes) ->
+    Picklock = application:get_env(?APP, ?PICKLOCK, <<>>),
```

Comment:
This is an absolute no-no. You cannot call `application:get_env/3` on the hot code path! All of those checks go through a single process in OTP, which makes this line the bottleneck for message throughput. This **must** be inferred in a different way instead of being recomputed every time.

Reply:
I get your point and totaly agree. I reanalized code and decided to move new functionality to check_token() method. There is another method - validate_payload() which also read from configuration file. I would love to have possibility to compute configuration parameters only once, for example when module is loaded, but I am not sure that Erlang could provide this.